File: LeetCode/2-Vector/26-removeDuplicates.cpp
//去除重复的元素，返回去重后数组的长度
//采用双指针的方式
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int p1 = 1, p2 = 1;
        int n = nums.size();
        if (n == 0) return 0;
        while (p1 < n)
        {
            if (nums[p1] != nums[p1-1])
            {
                nums[p2] = nums[p1];
                ++p2;
            }
            ++p1;
        }
        return p2;
    }
};

File: 面试高频leetcode/双指针/26_removeDuplicates.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        # 快慢指针
        if not nums:
            return 0
        fast = slow = 1
        while fast < len(nums):
            # 当遇到不同元素，慢指针接收快指针的元素
            if nums[fast] != nums[fast-1]:
                nums[slow] = nums[fast]
                slow += 1
            fast += 1
        return slow
